Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+
+# ####################################################################################################
+# Azure_Subscription_AuthZ_Remove_Deprecated_Accounts
+
+function Get-DeletedUser()
+{
+  <#
+    .SYNOPSIS
+    This command shows a deleted user.
+    .DESCRIPTION
+    This command shows a deleted user. This is helpful when trying to translate a scanner status message user GUID to a human display name in order to locate the user on a list of assignments.
+    This command requires the Microsoft.Graph SDK. See installation instructions: https://learn.microsoft.com/powershell/microsoftgraph/installation?view=graph-powershell-1.0#installation
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Key Vault.
+    .PARAMETER DeletedUserId
+    The user GUID for the deleted user.
+    .INPUTS
+    None
+    .OUTPUTS
+    None
+    .EXAMPLE
+    PS> Get-DeletedUser -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -DeletedUserId "00000000-xxxx-0000-xxxx-000000000000"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory=$true)]
+    [string]
+    $DeletedObjectId
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  Connect-MgGraph
+
+  Get-MgDirectoryDeletedItem -DirectoryObjectId $DeletedObjectId
+}
+
+# ####################################################################################################
+
+# ####################################################################################################
+# Azure_Subscription_DP_Avoid_Plaintext_Secrets_Deployments
+
+function Get-ResourceGroupDeployment()
+{
+  <#
+    .SYNOPSIS
+    This command retrieves ARM deployment(s) for the Resource Group.
+    .DESCRIPTION
+    This command retrieves ARM deployment(s) for the Resource Group.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID.
+    .PARAMETER ResourceGroupName
+    The Resource Group to which the Deployment was run.
+    .PARAMETER DeploymentName
+    Optional: a Deployment name. If not specified, all Resource Group deployments will be retrieved.
+    .INPUTS
+    None
+    .OUTPUTS
+    [PSResourceGroupDeployment](https://learn.microsoft.com/dotnet/api/microsoft.azure.commands.resourcemanager.cmdlets.sdkmodels.psresourcegroupdeployment)
+    .EXAMPLE
+    PS> Get-ResourceGroupDeployment -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DeploymentName "MyDeploymentName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$false)]
+      [string]
+      $DeploymentName = ""
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  if ( $DeploymentName )
```

Comment:
Shall we check for string null or whitespace, just in case user provides empty string with whitespaces?

Reply:
Good catch, flipped it around to if ([string]::IsNullOrWhiteSpace($DeploymentName)) to get all deployments, else just the provided one.